app/controllers/concerns/foreman/controller/provisioning_templates.rb

```diff
@@ -42,4 +43,17 @@ def process_template_kind
   def type_name_singular
     @type_name_singular ||= resource_class.to_s.underscore
   end
+
+  def global_registration_vars
+    permitted = Foreman::Plugin.all
+                               .map(&:allowed_registration_vars)
+                               .flatten.compact.uniq
+    {
+      user: User.current,
+      auth_token: User.current.jwt_token!(expiration: 4.hours.to_i),
+      organization: (Organization.current if params['organization_id']),
```

Comment:
This suggests that passing the organization ID is really a boolean. Same for the location.

Reply:
So what do you suggest?
Something like `Organization.current unless params['organization_id'].nil?)`

Reply:
Wouldn't it make more sense that if an org_id is given, register the host into org with that id, otherwise fallback to user's default org/current org? And the same for locations of course